### PR TITLE
Add tests which pass LLVM17 to CMake

### PR DESCRIPTION
**Pull request by advikkabra**

### CODE REVIEW

**Comment by certik** [comment]
Nice! Almost everything now compiles.

I merged some PRs, so you need to resolve the conflicts.

File: integration_tests/CMakeLists.txt [hunk 1]
```diff
@@ -1384,7 +1384,7 @@ RUN(NAME cpu_time_02_wasm LABELS wasm)
 RUN(NAME system_clock_01 LABELS gfortran llvm llvm17)
 RUN(NAME boz_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc llvm17)
 RUN(NAME cmd_01 LABELS gfortran llvm llvm_wasm llvm_wasm_emcc)
-RUN(NAME cmd_02 LABELS gfortran llvm NOFAST)
+RUN(NAME cmd_02 LABELS gfortran llvm)
```
Comment (Thirumalai-Shaktivel) [inline integration_tests/CMakeLists.txt:1387]: Revert this, the CI will pass

File: integration_tests/CMakeLists.txt [hunk 2]
```diff
@@ -1772,11 +1772,11 @@ RUN(NAME openmp_30 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp) # matmul
 RUN(NAME openmp_31 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp)
 RUN(NAME openmp_32 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp)
 RUN(NAME openmp_33 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp) # mandelbrot
-RUN(NAME openmp_34 LABELS gfortran llvm_omp GFORTRAN_ARGS -fopenmp) # Ensure uniform load distribution over threads
```
Comment (Thirumalai-Shaktivel) [inline integration_tests/CMakeLists.txt:1775]: Revert this.

**Review by Thirumalai-Shaktivel — COMMENTED** [review]
Yes, many tests are passing, we are almost done, thank you!

File: integration_tests/CMakeLists.txt (hunk 1 above)
Comment (advikkabra) [reply integration_tests/CMakeLists.txt:1387]: I'll do that. I think I removed that by mistake.

**Review by Thirumalai-Shaktivel — APPROVED** [review]

**Pull request merged**